src/test/java/core/basesyntax/HelloWorldTest.java

```diff
@@ -1,7 +1,189 @@
+
+class HelloWorldTest {
+    private static Reader reader;
+    private static DataConverter dataConverter;
+    private static ShopService shopService;
+    private static List<FruitTransaction> soloListFruit;
+    private static List<FruitTransaction> multiplyListFruits;
+    private static ReportGenerator reportGenerator;
+    private static Writer fileWriter;
+
+    @BeforeAll
+    public static void setUp() {
+        reader = new ReaderImpl();
+        dataConverter = new DataConverterImpl();
+        soloListFruit = new ArrayList<>();
+        soloListFruit.add(new FruitTransaction("b", "banana", 45));
+        multiplyListFruits = new ArrayList<>();
+
+        Map<FruitTransaction.Operation, OperationHandler> operationHandlers = new HashMap<>();
+        operationHandlers.put(FruitTransaction.Operation.BALANCE, new BalanceOperation());
+        operationHandlers.put(FruitTransaction.Operation.PURCHASE, new PurchaseOperation());
+        operationHandlers.put(FruitTransaction.Operation.RETURN, new ReturnOperation());
+        operationHandlers.put(FruitTransaction.Operation.SUPPLY, new SupplyOperation());
+        OperationStrategy operationStrategy = new OperationStrategyImpl(operationHandlers);
+        shopService = new ShopServiceImpl(operationStrategy);
+
+        reportGenerator = new ReportGeneratorImpl();
+        fileWriter = new WriterImpl();
+    }
+
+    @BeforeEach
+    public void setUpEach() {
+        soloListFruit.clear();
+        soloListFruit.add(new FruitTransaction("b", "banana", 45));
+    }
+
+    @Test
+    public void reader_correctFile_Ok() {
+        List<String> expected = new ArrayList<>();
+        expected.add("b,banana,45");
+        List<String> actual = reader.read("src/test/java/core/basesyntax/recources/inputFile.csv");
+        Assert.assertEquals("Readed data is wrong.", expected.toString(), actual.toString());
+    }
+
+    @Test
+    public void reader_incorrectFile_NotOk() {
+        Assert.assertThrows(RuntimeException.class, () -> {
+            reader.read("src/test/java/core/basesyntax/recources/incorrectFile.csv");
+        });
+    }
+
+    @Test
+    public void reader_emptyFile_NotOk() {
+        Assert.assertThrows(RuntimeException.class, () -> {
+            reader.read("src/test/java/core/basesyntax/recources/emptyFile.csv");
+        });
+    }
+
+    @Test
+    public void reader_incorrectData_NotOk() {
+        Assert.assertThrows(RuntimeException.class, () -> {
+            reader.read("src/test/java/core/basesyntax/recources/incorrectData.csv");
+        });
+    }
+
+    @Test
+    public void converter_correctData_Ok() {
+        List<String> data = reader.read("src/test/java/core/basesyntax/recources/inputFile.csv");
+        List<FruitTransaction> actual = dataConverter.convertToTransaction(data);
+        List<FruitTransaction> expected = soloListFruit;
+        Assert.assertEquals(1, actual.size());
+        Assert.assertEquals(actual.get(0), expected.get(0));
+    }
```

Comment:
separate your tests into descriptive blocks like 

`        
//given
List<String> data = reader.read("src/test/java/core/basesyntax/recources/inputFile.csv");
List<FruitTransaction> expected = soloListFruit;

//when
List<FruitTransaction> actual = dataConverter.convertToTransaction(data);
        
//then
Assert.assertEquals(1, actual.size());        
Assert.assertEquals(actual.get(0), expected.get(0));
`

Reply:
this relates to all tests